contracts/base/ModuleManager.sol

```diff
@@ -123,7 +123,7 @@ contract ModuleManager is SelfAuthorized, Executor {
             currentModule = modules[currentModule];
             moduleCount++;
         }
-        next = currentModule;
+        next = array[moduleCount == 0 ? 0 : moduleCount - 1];
```

Comment:
`array[0]` (for the case that `moduleCound == 0`)  would be undefined, right? In the best case it is 0 in the worst case it some random data (should normally not happen). 

I still believe we should first clearly define the expected behavior for this method (and maybe even for the .

- `start` has to be a module else the method should revert
- `pageSize` needs to be greater 0, else the method should revert 
- If all entries fit into a single page, then `next` should be Address One (as this is the current behavior)
- If another page is present then `next` should be the last element of the returned array (<- this was bugged before)

Reply:
> array[0] (for the case that moduleCound == 0) would be undefined, right? 

It's zero address I think

Reply:
yeah it *should* be zero (at least with the current solidity management), but not sure if there is a guarantee for it.

if `pageSize` is 0 this should revert as array access should do bounds checks.